### PR TITLE
Minor preview fixes

### PR DESCRIPTION
**Pull request by Aergwyn**
- prevent negative width on progress bar
- return the existing preview if it already exists instead of nothing which allows listening to it again

### CODE REVIEW

File: osu.Game/Overlays/BeatmapSet/PreviewButton.cs [hunk 1]
```diff
@@ -82,7 +83,8 @@ protected override void Update()
 
             if (Playing.Value && preview != null)
             {
-                progress.Width = (float)(preview.CurrentTime / preview.Length);
+                // prevent negative (potential infinite) width if a track without length was loaded
+                progress.Width = (float)Math.Max(preview.CurrentTime / preview.Length, 0);
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/PreviewButton.cs]: not sure `Math.Max` is the best way of handling this. how about a check on `Length > 0` instead?

**Review by peppy — DISMISSED** [review]

File: osu.Game/Overlays/BeatmapSet/PreviewButton.cs (hunk 1 above)
Comment (Aergwyn) [reply osu.Game/Overlays/BeatmapSet/PreviewButton.cs]: Instead this?
```
// prevent negative (potential infinite) width if a track without length was loaded
if(preview.Length > 0)
  progress.Width = (float)preview.CurrentTime / preview.Length;
else
  progress.Width = 0;
```
May be a bit more verbose but also easier to read.

Comment (peppy) [reply osu.Game/Overlays/BeatmapSet/PreviewButton.cs]: can probably use tertiary ?: but yeah, that sounds fine.

**Review by smoogipoo — APPROVED** [review]
Seems good 👍

**Pull request merged**